driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,246 @@
+		}
+		repeatType := settings.Repeat
+		nodename := settings.Nodename
+		epnum := settings.Epnum
+
+		// Build Schedule by Repeat Type
+		if repeatType == "Daily" {
+			schedule, scheduleError := getScheduleByDay(0, epnum, sitename, nodename, cookie, pel.timezone)
+			if scheduleError != nil {
+				return nil, fmt.Errorf("Error retrieving schedule for thermostat %v: %v", nodename, scheduleError)
+			}
+			for _, day := range week {
+				thermSchedule.DaySchedules[day] = schedule
+			}
+		} else if repeatType == "Weekly" {
+			for index, day := range week {
+				schedule, scheduleError := getScheduleByDay(index, epnum, sitename, nodename, cookie, pel.timezone)
+				if scheduleError != nil {
+					return nil, fmt.Errorf("Error retrieving schedule for thermostat %v on %v (day %v): %v", nodename, day, index, scheduleError)
+				}
+				thermSchedule.DaySchedules[day] = schedule
+			}
+		} else if repeatType == "Weekday/Weekend" {
+			weekend, weekendError := getScheduleByDay(0, epnum, sitename, nodename, cookie, pel.timezone)
+			if weekendError != nil {
+				return nil, fmt.Errorf("Error retrieving schedule for thermostat %v on weekend (day 0): %v", nodename, weekendError)
+			}
+			for _, day := range []string{"Sunday", "Saturday"} {
+				thermSchedule.DaySchedules[day] = weekend
+			}
+			weekday, weekdayError := getScheduleByDay(1, epnum, sitename, nodename, cookie, pel.timezone)
+			if weekdayError != nil {
+				return nil, fmt.Errorf("Error retrieving schedule for thermostat %v on weekday (day 1): %v", nodename, weekdayError)
+			}
+			for _, day := range []string{"Monday", "Tuesday", "Wednesday", "Thursday", "Friday"} {
+				thermSchedule.DaySchedules[day] = weekday
+			}
+		} else {
+			return nil, fmt.Errorf("Failed to recognize repeat type of thermostat %v's schedule: %v", nodename, repeatType)
+		}
+
+		schedules[child.Id] = thermSchedule
+	}
+	return schedules, nil
+}
+
+func getSettings(sitename, thermostatID string, cookie *http.Cookie) (settingsWrapper, error) {
+	var requestURL bytes.Buffer
+	requestURL.WriteString(fmt.Sprintf("https://%s.officeclimatecontrol.net/ajaxThermostat.cgi?id=", sitename))
+	requestURL.WriteString(thermostatID)
+	requestURL.WriteString(":Thermostat&request=GetSchedule")
+
+	resp, _, errs := gorequest.New().Get(requestURL.String()).Type("form").AddCookie(cookie).End()
+	if errs != nil {
+		return settingsWrapper{}, fmt.Errorf("Failed to retrieve schedule settings for thermostat %v: %v", thermostatID, errs)
+	}
+	var result settingsRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return settingsWrapper{}, fmt.Errorf("Failed to decode schedule settings for thermostat %v: %v", thermostatID, decodeError)
+	}
+	return result.Userdata, nil
+}
+
+func getScheduleByDay(dayOfWeek int, epnum int64, sitename, thermostatID string, cookie *http.Cookie, timezone *time.Location) (ThermostatDaySchedule, error) {
```

Comment:
Have you thought about returning a pointer to a `ThermostatDaySchedule` instead? Then when you have an error you can just return `nil` instead of an empty struct. Should be a bit cleaner.

Similar idea for the `getSettings` function as well.

Reply:
Good point. I modified the return types and variables of getScheduleByDay and getSettings to be pointers to structs, not the structs themselves.